File: room-demo-rxandroid-livedata/src/main/java/com/example/pavneet_singh/room_demo_rxandroid/adapter/NotesAdapter.java
package com.example.pavneet_singh.room_demo_rxandroid.adapter;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.pavneet_singh.room_demo_rxandroid.R;
import com.example.pavneet_singh.room_demo_rxandroid.notedb.model.Note;

import org.jetbrains.annotations.NotNull;

import java.util.List;


/**
 * Created by Pavneet_Singh on 12/20/17.
 */

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.BeanHolder> {

    private final List<Note> list;
    private final LayoutInflater layoutInflater;
    private final OnNoteItemClick onNoteItemClick;

    public NotesAdapter(List<Note> list, Context context) {
        layoutInflater = LayoutInflater.from(context);
        this.list = list;
        this.onNoteItemClick = (OnNoteItemClick) context;
    }


    @NotNull
    @Override
    public BeanHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        View view = layoutInflater.inflate(R.layout.note_list_item, parent, false);
        return new BeanHolder(view);
    }

    @Override
    public void onBindViewHolder(BeanHolder holder, int position) {
        Log.e("bind", "onBindViewHolder: " + list.get(position));
        holder.textViewTitle.setText(list.get(position).getTitle());
        holder.textViewContent.setText(list.get(position).getContent());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class BeanHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        final TextView textViewContent;
        final TextView textViewTitle;

        BeanHolder(View itemView) {
            super(itemView);
            itemView.setOnClickListener(this);
            textViewContent = itemView.findViewById(R.id.item_text);
            textViewTitle = itemView.findViewById(R.id.tv_title);
        }

        @Override
        public void onClick(View view) {
            onNoteItemClick.onNoteClick(getAdapterPosition());
        }
    }

    public interface OnNoteItemClick {
        void onNoteClick(int pos);
    }
}

File: room-demo-kotlin-mvvm-dagger/src/main/java/com/example/pavneet_singh/room_demo_kotin_mvvm_dagger/adapter/NotesAdapter.kt
package com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.R
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.base.BaseBindViewHolder
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.base.BindingAdapterI
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.databinding.NoteListItemBinding
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.notedb.model.Note

/**
 * Created by Pavneet_Singh on 2020-01-25.
 */

class NotesAdapter(
        private var list: List<Note>,
        context: Context
) :
        RecyclerView.Adapter<NotesAdapter.NoteHolder?>(), BindingAdapterI<List<Note>> {
    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private val onNoteItemClick: OnNoteItemClick
    override fun onCreateViewHolder(
            parent: ViewGroup,
            viewType: Int
    ): NoteHolder {
        return NoteHolder(
                DataBindingUtil.inflate(
                        layoutInflater,
                        R.layout.note_list_item,
                        parent,
                        false
                )
        )
    }

    override fun onBindViewHolder(
            holder: NoteHolder,
            position: Int
    ) = holder.bind(list[position])

    override fun getItemCount(): Int {
        return list.size
    }

    inner class NoteHolder(private val binding: NoteListItemBinding) :
            BaseBindViewHolder<Note>(binding.root) {

        init {
            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View?) {
            onNoteItemClick.onNoteClick(list[adapterPosition])
        }

        override fun bind(model: Note) {
            binding.note = model
        }

    }

    interface OnNoteItemClick {
        fun onNoteClick(note: Note)
    }

    init {
        onNoteItemClick = context as OnNoteItemClick
    }

    override fun setSource(data: List<Note>) {
        list = data
        notifyDataSetChanged()
    }
}